### PR TITLE
python3Packages.uproot: init at 4.0.8

### PR DESCRIPTION
**Pull request by veprbl**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by veprbl** [comment]
@GrahamcOfBorg build python38Packages.uproot python39Packages.uproot

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 124557` at e5ecf524 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>python38Packages.scikit-hep-testdata</li>
    <li>python38Packages.uproot</li>
    <li>python39Packages.scikit-hep-testdata</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/scikit-hep-testdata/default.nix:48:0:
```
   |
48 |     description = "A common package to provide example files (e.g., ROOT) for testing and developing packages against.";
   | ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/scikit-hep-testdata/default.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  checkInputs = [
+    pytestCheckHook
+  ];
+  preCheck = ''
+    export HOME="$(mktemp -d)"
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/scikit-hep/scikit-hep-testdata";
+    description = "A common package to provide example files (e.g., ROOT) for testing and developing packages against.";
```
Comment (fabaff) [inline pkgs/development/python-modules/scikit-hep-testdata/default.nix:48]: No period at the end, see https://nixos.org/manual/nixpkgs/stable/#sec-standard-meta-attributes

```suggestion
    description = "Common package to provide example files (e.g., ROOT) for testing and developing packages against";
```

Comment (veprbl) [reply pkgs/development/python-modules/scikit-hep-testdata/default.nix]: Removed the period, left the article.

**Comment by veprbl** [comment]
@GrahamcOfBorg build python38Packages.uproot python39Packages.uproot

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 124557` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python38Packages.scikit-hep-testdata</li>
    <li>python38Packages.uproot</li>
    <li>python39Packages.scikit-hep-testdata</li>
  </ul>
</details>

File: pkgs/development/python-modules/scikit-hep-testdata/default.nix [hunk 2]
```diff
@@ -0,0 +1,53 @@
+{ lib
+, fetchFromGitHub
+, python
+, buildPythonPackage
+, importlib-resources
+, pyyaml
+, requests
+, pytestCheckHook
+, setuptools_scm
```
Comment (fabaff) [inline pkgs/development/python-modules/scikit-hep-testdata/default.nix:9]: The old identifier is still working but for new packages we should stick to the naming schema according [PEP 0503](https://nixos.org/manual/nixpkgs/stable/#contributing).

```suggestion
, setuptools-scm
```

File: pkgs/development/python-modules/scikit-hep-testdata/default.nix [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  };
+
+  nativeBuildInputs = [
+    git
+  ];
+  buildInputs = [
+    setuptools_scm
+  ];
+  propagatedBuildInputs = [
+    importlib-resources
```
Comment (fabaff) [inline pkgs/development/python-modules/scikit-hep-testdata/default.nix:37]: For the latest Python releases we should use the `importlib.resources` module which is part of the Python standard library.

Replace `importlib-resources` with

```suggestion
     ++ lib.optionals (pythonOlder "3.9") [ importlib-resources ];
```

File: pkgs/development/python-modules/scikit-hep-testdata/default.nix [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+, pyyaml
+, requests
+, pytestCheckHook
+, setuptools_scm
+, git
+}:
+
+buildPythonPackage rec {
+  pname = "scikit-hep-testdata";
+  version = "0.4.2";
```
Comment (fabaff) [inline pkgs/development/python-modules/scikit-hep-testdata/default.nix:15]: In the meantime a [new release](https://github.com/scikit-hep/scikit-hep-testdata/releases/tag/v0.4.3) was published.

**Comment by veprbl** [comment]
@GrahamcOfBorg build python38Packages.uproot python39Packages.uproot

**Comment by veprbl** [comment]
@GrahamcOfBorg build python38Packages.uproot python39Packages.uproot

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 124557` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python38Packages.scikit-hep-testdata</li>
    <li>python38Packages.uproot</li>
    <li>python39Packages.scikit-hep-testdata</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 124557` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>python38Packages.uproot: log was empty</li>
  </ul>
</details>
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>python38Packages.scikit-hep-testdata: <a href="https://termbin.com/z3do">plain log</a>  |  <a href="https://htmlpreview.github.io/?https://gist.githubusercontent.com/SuperSandroBot/1d331cd1a64a65214db923ec7fa52a12/raw/index.html">pretty log </a></li>
  </ul>
</details>
<details>
<summary>
1 package built:
</summary>
<ul>
<li>python39Packages.scikit-hep-testdata</li>
</ul>
</details>

**Pull request merged**